src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/DynamicContentTypeBuilder.cs

```diff
@@ -75,6 +76,13 @@ public void Build(ISchema schema, FieldType contentQuery, ContentTypeDefinition
                                     continue;
                                 }
 
+                                // Do not add field if it collides with existing ones. Note that we may have fields with different
+                                // casing here too, which must be prevented as well.
+                                if (contentItemType.Fields.Any(f => f.Name.Equals(fieldType.Name, StringComparison.OrdinalIgnoreCase)))
```

Comment:
This logic has three copies. Create a helper class or extension method for it instead.

Reply:
Done.

Reply:
Don't repeat the comments either.

Reply:
I'd rather leave the comments to make sure nobody replaces the class with the GraphQL `HasField` method. Should I?

Reply:
I think if one attempts to replace the calls, they'll first check out what they're actually replacing (wishful thinking?) so a comment is adequate only there.

Reply:
I removed the redundant comment in one file.

Reply:
Why not all?

Reply:
Because it's in different files...

Reply:
I mean, you had three copies of the logic. Now you have one method, and three consuming code of it. But the comment was still copied to three places. You removed one, and now it's only in two places. But why? :) All I'm saying is just have that comment on the central method, because everything is using just that.

Reply:
The comment is not about the method or its logic, but about the call site and why it doesn't use the native GraphQL method `HasField`. IMHO it doesn't make sense on the method itself. I'm going to remove it, if that makes you happy.

Reply:
The comment

```
// Do not add field if it collides with existing ones. Note that we may have fields with different
// casing here too, which must be prevented as well.
```

is about the method, and thus should only be on the method. Adding a comment about why a given code uses this method and not `HasField` (something you didn't mention before) is indeed about the call site, and should be added to the call site. But that's a different comment. So, please add that, then.

It's not about making me happy but having good code in OC. Having good code in OC does make me happy, so perhaps it's all about my happiness, in the end? :)

Reply:
Even though I find it super annoying having to discuss this nitpicking, I didn't intent to be offensive, sorry. I really hope it makes you happy 😊

But the comments were all about the call sites from the beginning, it's not about the method. I thought that it clearly explains why the if condition is there and that fields can have different casing that should not be added as well. Read it with the whole context:

```C#
// Do not add field if it collides with existing ones. Note that we may have fields with different
// casing here too, which must be prevented as well.
if (contentItemType.HasFieldIgnoreCase(fieldType.Name))
{
    continue;
}
```

How would you phrase it?

Reply:
Like this:

```c#
    /// <summary>
    /// Checks if the field exists in the graph type in a case-insensitive way.
    /// </summary>
    /// <remarks>
    /// <para>
    /// This is the same as calling <see cref="IComplexGraphType.HasField(string)"/> but in a case-insensitive way. OC
    /// fields may be added with different casings, and we want to avoid collisions even then.
    /// </para>
    /// <para>
    /// See <see href="https://github.com/OrchardCMS/OrchardCore/pull/16151"/> and its corresponding issues for context.
    /// </para>
    /// </remarks>
    public static bool HasFieldIgnoreCase(this IComplexGraphType graphType, string fieldName)
        => graphType.Fields.Any(field => field.Name.Equals(fieldName, StringComparison.OrdinalIgnoreCase));
```

But feel free to keep the other comments as well if you think they add clarity.

Reply:
Thanks @Piedone, I added your comment.